app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -210,23 +210,27 @@ class HistoryFragment : Fragment(), CoroutineScope, BackHandler {
     override fun onBackPressed(): Boolean = (historyComponent.uiView as HistoryUIView).onBackPressed()
 
     private suspend fun reloadData() {
-        val allowedVisitTypes = listOf(VisitType.LINK, VisitType.TYPED, VisitType.BOOKMARK)
+        val excludeTypes = listOf(
+            VisitType.NOT_A_VISIT,
+            VisitType.DOWNLOAD,
```

Comment:
Not sure about excluding these. IIUC, typing a file url directly (that will auto-download) may be tagged as "download"?